src/services/YotiService.ts

```diff
@@ -0,0 +1,341 @@
+/* eslint-disable no-console */
+import { Logger } from "@aws-lambda-powertools/logger";
+import crypto from "crypto";
+import { v4 as uuidv4 } from "uuid";
+import axios, { AxiosRequestConfig } from "axios";
+import { HttpVerbsEnum } from "../utils/HttpVerbsEnum";
+import { PersonIdentity } from "../models/PersonIdentity";
+import { StructuredPostalAddress, ApplicantProfile, PostOfficeInfo, SessionInfo } from "../models/yotiPayloads";
+
+export class YotiService {
+  readonly logger: Logger;
+
+  private static instance: YotiService;
+
+  readonly CLIENT_SDK_ID: string;
+
+  readonly PEM_KEY: string;
+
+  constructor(logger: Logger, CLIENT_SDK_ID: string, PEM_KEY: string) {
+  	this.logger = logger;
+  	this.CLIENT_SDK_ID = CLIENT_SDK_ID;
+  	this.PEM_KEY = PEM_KEY;
+  }
+
+  static getInstance(
+  	logger: Logger,
+  	CLIENT_SDK_ID: string,
+  	PEM_KEY: string,
+  ): YotiService {
+  	if (!YotiService.instance) {
+  		YotiService.instance = new YotiService(logger, CLIENT_SDK_ID, PEM_KEY);
+  	}
+  	return YotiService.instance;
+  }
+
+  private getRSASignatureForMessage(message: string): string {
+  	return crypto
+  		.createSign("RSA-SHA256")
+  		.update(message)
+  		.sign(this.PEM_KEY)
+  		.toString("base64");
+  }
+
+
+  //TODO: Hardcoded now, will update with values from Person Identity table once /session work completed
+  private getStructuredPostalAddress(personDetails: PersonIdentity): StructuredPostalAddress {
+  	return {
+  		address_format: 1,
+  		building_number: "74",
+  		address_line1: "AddressLine1",
+  		town_city: "CityName",
+  		postal_code: "E143RN",
+  		country_iso: "GBR",
+  		country: "United Kingdom",
+  	};
+  }
+
+  private getApplicantProfile(
+  	personDetails: PersonIdentity,
+  ): ApplicantProfile {
+  	const givenNames: string[] = [];
+  	const familyNames: string[] = [];
+
+  	for (const name of personDetails.names) {
+  		const nameParts = name.nameParts;
+  		for (const namePart of nameParts) {
+  			if (namePart.type === "GivenName") {
+  				givenNames.push(namePart.value);
+  			}
+  			if (namePart.type === "FamilyName") {
+  				familyNames.push(namePart.value);
+  			}
+  		}
+  	}
+
+  	return {
+  		full_name: `${givenNames[0]} ${familyNames}`,
+  		date_of_birth: `${personDetails.birthDates.map((bd) => ({ value: bd.value }))[0].value}`,
+  		structured_postal_address:
+        this.getStructuredPostalAddress(personDetails),
+  	};
+  }
+
+  private getYotiDocumentType(selectedDocument: string): {
+  	yotiDocumentType: string;
+  	countryCode: string;
+  } {
+  	let yotiDocumentType = "";
+  	let countryCode = "";
+
+  	switch (selectedDocument) {
+  		case "ukPassport": {
```

Comment:
can this be moved as enums to match F2f document type to yotiDocumentType

Reply:
there is ongoing discussion on how to get country code, best it to keep it as GBR in  Constants and use it here